### PR TITLE
HttpRequestFirstLine::setMethod() bugfix

### PR DESCRIPTION
**Pull request by Roman-Koshelev**
Repeated method invocation spoils the query string. eg "POSTTTT", "PATCHCHCHCH", "CONNECTNECTNECTNECT", "OPTIONSIONSIONSIONS".

### CODE REVIEW

**Comment by seladb** [comment]
It seems I simply missed setting `m_Method` in `setMethod()`, right? 

Could you please explain why it spoil the query string?

**Comment by Roman-Koshelev** [comment]
string `int lengthDifference = MethodEnumToString[newMethod].length() - MethodEnumToString[m_Method].length();` . When calling a method many times with the same lengthDifference value, it should be 0.

**Comment by seladb** [comment]
It seems the same bug exists also in:

https://github.com/seladb/PcapPlusPlus/blob/5937d3c8acfeaf5581cd248bdcf2f2a8a960fdb0/Packet%2B%2B/src/HttpLayer.cpp#L486-L496

and in:

https://github.com/seladb/PcapPlusPlus/blob/5937d3c8acfeaf5581cd248bdcf2f2a8a960fdb0/Packet%2B%2B/src/HttpLayer.cpp#L855-L862

Maybe we should fix it there too?

Also, it'd be great to add some unit-tests to make sure the bug is solved. You can add a few tests here:

For `setMethod()`:

https://github.com/seladb/PcapPlusPlus/blob/f25cf80f5fe4c5f9b837edfda775c57e04585d36/Tests/Packet%2B%2BTest/main.cpp#L2107

For `setVersion()`:

https://github.com/seladb/PcapPlusPlus/blob/f25cf80f5fe4c5f9b837edfda775c57e04585d36/Tests/Packet%2B%2BTest/main.cpp#L2302

**Comment by Roman-Koshelev** [comment]
setVersion at least doesn't spoil the query string)

File: Tests/Packet++Test/main.cpp [hunk 1]
```diff
@@ -2105,6 +2105,9 @@ PTF_TEST_CASE(HttpRequestLayerCreationTest)
 	PTF_ASSERT(httpPacket.addLayer(&httpLayer), "Adding HTTP request layer failed");
 	hostField->setFieldValue("www.ynet.co.il");
 	httpLayer.getFirstLine()->setMethod(HttpRequestLayer::HttpGET);
+	PTF_ASSERT(httpLayer.getFirstLine()->getMethod() == HttpRequestLayer::HttpGET, "Couldn't set method");
+	httpLayer.getFirstLine()->setVersion(pcpp::HttpVersion::OneDotOne);
+	PTF_ASSERT(httpLayer.getFirstLine()->getVersion() == pcpp::HttpVersion::OneDotOne, "Couldn't set version");
```
Comment (seladb) [inline Tests/Packet++Test/main.cpp]: these doesn't compile as you can see here:
https://travis-ci.org/seladb/PcapPlusPlus/jobs/603934668#L325

, please `pcpp::HttpVersion::OneDotOne` with `OneDotOne`

**Comment by Roman-Koshelev** [comment]
done

**Pull request merged**